returnn/tf/layers/basic.py

```diff
@@ -625,24 +625,16 @@ def __init__(self, axes, param_shape="F", scale=True, bias=True, epsilon=1e-6, *
     super(NormLayer, self).__init__(**kwargs)
     assert not self.input_data.sparse
     x = self.input_data.placeholder
-    if scale or bias:
-      if isinstance(param_shape, int):
-        param_shape = [param_shape]
-      elif ((isinstance(param_shape, list) or isinstance(param_shape, tuple)) and
-            all(isinstance(x, int) for x in param_shape)):
-        param_shape = param_shape
-      elif (isinstance(param_shape, str) or
-            ((isinstance(param_shape, list) or isinstance(param_shape, tuple)) and
-            all(isinstance(x, str) for x in param_shape))):
-        param_shape = self.input_data.get_bc_shape(opts={ax: -1 for ax in param_shape})
-      assert len(param_shape) == len(self.input_data.batch_shape), (
-        "Dimension does not match input. Maybe broadcast dimensions are not included explicitly?")
-      assert all(isinstance(x, int) for x in param_shape)
+    assert isinstance(param_shape, str)  # not implemented otherwise yet
+    param_axes = sorted(self.input_data.get_axes_from_description(param_shape))
+    param_shape = [self.input_data.batch_shape[axis] for axis in param_axes]
+    assert all(isinstance(dim, int) for dim in param_shape), "%s: only static param shape allowed" % self
+    param_bc_shape = [dim if axis in param_axes else 1 for (axis, dim) in enumerate(self.input_data.batch_shape)]
     axes = self.input_data.get_axes_from_description(axes)
 
     with self.var_creation_scope():
-      scale_params = self.add_param(tf_compat.v1.get_variable("scale", param_shape, initializer=tf.ones_initializer()))
-      bias_params = self.add_param(tf_compat.v1.get_variable("bias", param_shape, initializer=tf.zeros_initializer()))
+      scale_params = self.add_param(tf_compat.v1.get_variable("scale", param_bc_shape, initializer=tf.ones_initializer()))
+      bias_params = self.add_param(tf_compat.v1.get_variable("bias", param_bc_shape, initializer=tf.zeros_initializer()))
```

Comment:
The param itself should not have the broadcasted shape. It should just use `param_shape`. You add the broadcast dims afterwards (via `tf.reshape`).

Reply:
Also singular, `scale_param`, `bias_param`.
Also, this should be conditional, only if `scale`, and only if `bias` (two separate `if`s).

Reply:
Actually, you already have the `if` below. Just move this code over there.